### PR TITLE
docs: more flowctl guides

### PR DESCRIPTION
**Pull request by oliviamiannone**
**Description:**

This PR adds a new flowctl-specific node to the Guides section of the docs and populates it with:
* The existing derivation guide (moved)
* Three new guides suggested [in this discussion](https://github.com/estuary/flow/pull/859#issuecomment-1369405044)

plus relevant links and conceptual changes

Also adds a [redirect plugin](https://docusaurus.io/docs/api/plugins/@docusaurus/plugin-client-redirects) to accommodate the moved guide and all future docs pages that move. Note that it only works in production so I haven't been able to verify yet that it works. 

**Notes for reviewers:**

* For all references to authentication, I'm pointing to the `auth token --token` method still, because `auth login` doesn't work on mac... at least on mine.
* Previously in the flowctl docs (mainly the concepts page), I'd been referring to what we internally call a "draft" as a "synced draft" and files pulled locally as a "local draft." This time around I found that to be pretty clunky so I changed it. Now drafts are just drafts, and local source files are just local source files.

- - -
This change is [Reviewable](https://reviewable.io/reviews/estuary/flow/897)

### CODE REVIEW

File: site/docs/guides/flowctl/edit-specification-locally.md
Comment (psFried) [inline site/docs/guides/flowctl/edit-specification-locally.md:86]:
```suggestion
  By default, `flowctl` will put the specs into subdirectories organized by the names of the entities.
```

This wording seems a little too vague to me.

File: site/docs/guides/flowctl/edit-specification-locally.md
Comment (psFried) [inline site/docs/guides/flowctl/edit-specification-locally.md:106]: nit: the `test` command is checking a lot more than just the formatting. And IDK if "generic" is pulling much weight:

```suggestion
3. As you work, you can run `flowctl catalog test --source flow.yaml` to test your changes.
```

File: site/docs/guides/flowctl/troubleshoot-task.md
Comment (psFried) [inline site/docs/guides/flowctl/troubleshoot-task.md:33]: This may output logs from significantly _before_ the requested start time, but the requested start time will always be covered by the logs that were output. It kinda feels like we should try to clarify that :shrug:

File: site/docs/guides/flowctl/create-derivation.md
Comment (psFried) [inline site/docs/guides/flowctl/create-derivation.md:40]: I think these steps should change for the latest flowctl. There's no longer a need to use a draft in order to pull specs, and there's also no longer a need to pull the source collection specs in the first place. Pretty much all of the steps between here and step 3 of `Add a derivation` are no longer necessary. And I think the stuff about finding the specific file to put your derivation into should perhaps be replaced with a slightly more generic instruction of how to name the file based on the desired name of your derivation, since there likely won't be existing specs in their directory. Totally fine to do this in a subsequent PR, though.

**Review by psFried — APPROVED** [review]
This is a nice improvement. I left some comments inline, but should be fine to merge once those are addressed. The "create a derivation" guide I think still needs a bit of an overhaul, though I don't think it needs to hold up merging the rest of this.

File: site/docs/guides/flowctl/edit-draft-from-webapp.md
Comment (psFried) [reply site/docs/guides/flowctl/edit-draft-from-webapp.md:56]: I think I spoke too soon here. Upon a closer look, it doesn't seem like the benefit justifies the work at this point, so I'm holding off on any changes there.

File: site/docs/guides/flowctl/create-derivation.md
Comment (oliviamiannone) [reply site/docs/guides/flowctl/create-derivation.md:40]: Noted - I wasn't looking at this page too hard here but I should do so soon, and will make these changes at that point.

**Pull request merged**
